### ISSUE TITLE
Revisiting the issue of whether to transpose the population_predict output

### ISSUE THREAD

**Issue by jgabry** [mitzimorris/mrp-kit#101]
Right now `population_predict` transposes the probabilities so that the returned matrix is poststrat cells x draws. We then sum over the columns in the aggregate method. We could, alternatively, not transpose and then sum over rows. We should figure out if there's a meaningful speed/memory different between the two approaches.

**Comment by lauken13** [comment]
I think even if there's no speed difference (and if there is I suspect it would be the current way), we should keep it in it's current form because it's tidy.

**Comment by jgabry** [comment]
> we should keep it in it's current form because it's tidy.

I don't think this really fits into the "tidy" paradigm whether we transpose it or not. To be "tidy" we'd need (quoting from the [tidyr vignette](https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html#tidy-data))

> 1. Every column is a variable.
> 2. Every row is an observation.
> 3. Every cell is a single value

Even though what we have doesn't perfectly fit this definition either way, it actually seems to me that the way we have it now is _less_ tidy. The closest thing to multiple observations we have (in this particular object) is multiple draws, in which case to be tidy they should be the rows. But we have them as the columns. Or I'm totally misunderstanding "tidy", which is entirely possible!

**Comment by lauken13** [comment]
Yes, sorry you're right there. What I meant was that it's only one pivot_longer away from being tidy. 

In the rest of the code we combine it with the poststrat matrix (cbind), and then use group_by to do population and sub population estimates. Although I'm aware that row wise operations are now possible with the tidyverse, I'm not sure that it's advised to use them when given a choice between the two.

**Comment by jgabry** [comment]
> Although I'm aware that row wise operations are now possible with the tidyverse, I'm not sure that it's advised to use them when given a choice between the two.

Yeah I'm not sure, but I'd guess you're right that it's not advisable if there's a choice

**Comment by lauken13** [comment]
Is there speed testing tools in R? I only know of tictoc. It's probably worth checking I guess.

**Comment by jgabry** [comment]
Yeah there are lots of them. One I like is microbenchmark, but tictoc could be better (I haven't used it). There's also profiling, which involves tracking the memory and speed at each step in the code. For that I've used profviz but there may be other options.

**Comment by lauken13** [comment]
Options:

- apply matrix[subset, ]. Also a cost for transpose
- apply matrix[, subset] columns = poststrat_levels, rows = draws, 
- pivot_longer and use group_by and summarise
- something using data.table?
